src/link.js

```diff
@@ -113,11 +95,29 @@ module.exports = function link(config, args, callback) {
   };
 
   const tasks = dependencies.map((dependency) => (next) => {
-    const prelink = makeHook(dependency, 'prelink');
-    const postlink = makeHook(dependency, 'postlink');
-    const link = makeLink(dependency);
+    const queue = [];
+
+    var prelink;
+    var postlink;
+
+    if (dependency.config.hooks) {
+      prelink = dependency.config.hooks.prelink;
+      postlink = dependency.config.hooks.postlink;
+    }
```

Comment:
if we don't have any hooks, the code above will fail (Can't get prelink of undefined). That's why I wrapped it into check.